File: packages/sdk-java/src/main/java/com/ubag/sdk/UbagTransportException.java
package com.ubag.sdk;

/** Thrown when a request could not be sent (network/transport failure). */
public class UbagTransportException extends RuntimeException {

    private final String method;
    private final String url;

    public UbagTransportException(String method, String url, Throwable cause) {
        super("UBAG API request could not be sent: " + method + " " + url, cause);
        this.method = method;
        this.url = url;
    }

    public String method() {
        return method;
    }

    public String url() {
        return url;
    }
}

File: packages/sdk-csharp/Ubag.Sdk/UbagTransportException.cs
namespace Ubag.Sdk;

/// <summary>Thrown when a request could not be sent (network/transport failure).</summary>
public sealed class UbagTransportException : Exception
{
    public UbagTransportException(string httpMethod, string url, Exception innerException)
        : base($"UBAG API request could not be sent: {httpMethod} {url}: {innerException.Message}", innerException)
    {
        HttpMethod = httpMethod;
        Url = url;
    }

    public string HttpMethod { get; }

    public string Url { get; }
}
